Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,494 @@
+    ];
+
+    /**
+     * @var array
+     */
+    protected $menuConfig = [
+        'special' => 'language',
+        'wrap' => '[|]'
+    ];
+
+    /**
+     * @var array
+     */
+    protected $menuLevelConfig = [
+        'doNotLinkIt' => '1',
+        'wrapItemAndSub' => '{|}, |*| {|}, |*| {|}',
+        'stdWrap.' => [
+            'cObject' => 'COA',
+            'cObject.' => [
+                '10' => 'LOAD_REGISTER',
+                '10.' => [
+                    'languageUid.' => [
+                        'cObject' => 'TEXT',
+                        'cObject.' => [
+                            'value' => '', // Will be set by prepareConfiguration
+                            'listNum.' => [
+                                'stdWrap.' => [
+                                    'data' => 'register:count_HMENU_MENUOBJ',
+                                    'wrap' => '|-1'
+                                ],
+                                'splitChar' => ','
+                            ]
+                        ]
+                    ]
+                ],
+                '20' => 'USER',
+                '20.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'title',
+                    'stdWrap.' => [
+                        'wrap' => '"title":|'
+                    ]
+                ],
+                '21' => 'TEXT',
+                '21.' => [
+                    'value' => self::LINK_PLACEHOLDER,
+                    'wrap' => ',"link":|',
+                ],
+                '22' => 'TEXT',
+                '22.' => [
+                    'value' => self::TARGET_PLACEHOLDER,
+                    'wrap' => ',"target":|',
+                ],
+                '30' => 'TEXT',
+                '30.' => [
+                    'value' => '0',
+                    'wrap' => ',"active":|'
+                ],
+                '40' => 'TEXT',
+                '40.' => [
+                    'value' => '0',
+                    'wrap' => ',"current":|'
+                ],
+                '50' => 'TEXT',
+                '50.' => [
+                    'value' => '1',
+                    'wrap' => ',"available":|'
+                ],
+                '60' => 'TEXT',
+                '60.' => [
+                    'stdWrap.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'wrap' => ',"languageUid":"|"'
+                ],
+                '70' => 'USER',
+                '70.' => [
+                    'userFunc' => 'TYPO3\CMS\Frontend\DataProcessing\LanguageMenuProcessor->getFieldAsJson',
+                    'language.' => [
+                        'data' => 'register:languageUid'
+                    ],
+                    'field' => 'nav_title',
```

Comment:
still unsure about the nav_title thing, in case of we need it as separeate field - because would not like to change templates and write fallbacks when nav_title is not set in my template - i would like to see a field i can rely on that this one is filled with either nav_title or title.

Reply:
i already integrated a fallback see https://github.com/gilbertsoft/bootstrap_package/blob/c3e0dd3297ac60ec6c4bdfa49d100640c02cf156/Classes/Utility/LanguageUtility.php#L84. i think having two fields is more flexible and with the fallback it should run out of the box

Reply:
ok, cool - then everything is fine for me.

Reply:
please close change request